File: chazm-model/src/main/java/runtimemodels/chazm/model/event/CharacteristicEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.entity.Characteristic;

/**
 * The {@linkplain CharacteristicEventFactory} interface defines the API for constructing {@linkplain CharacteristicEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface CharacteristicEventFactory {

    /**
     * Constructs an {@linkplain CharacteristicEvent}.
     *
     * @param category       the {@linkplain EventType}.
     * @param characteristic the {@linkplain Characteristic}.
     * @return a {@linkplain CharacteristicEvent}.
     */
    CharacteristicEvent build(EventType category, Characteristic characteristic);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/CharacteristicEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.entity.Characteristic

/**
 * The [CharacteristicEventFactory] interface defines the API for constructing [CharacteristicEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface CharacteristicEventFactory {

    /**
     * Constructs an [CharacteristicEvent].
     *
     * @param category       the [EventType].
     * @param characteristic the [Characteristic].
     * @return a [CharacteristicEvent].
     */
    fun build(category: EventType, characteristic: Characteristic): CharacteristicEvent

}
